turbinia/workers/analysis/macho.py

```diff
@@ -0,0 +1,600 @@
+    Args:
+      binary (lief.MachO.Binary): binary to be parsed.
+      result (TurbiniaTaskResult): The object to place task results into.
+    Returns:
+      List[str]: List of the symbols.
+    """
+    symbols = []
+    #result.log(f'----------- symbols --------------')
+    for sym in binary.symbols:
+      symbols.append(sym.demangled_name)
+    #result.log(f'-----------------------------------')
+    return symbols
+
+  def _CSHashType(self, cs_hash_type):
+    """Translates CS hash type to a string.
+    Args:
+      cs_hash_type (int): CS hash type.
+    Returns:
+      (str): CS hash type string.
+    """
+    if cs_hash_type == 1:
+      return "SHA1"
+    elif cs_hash_type == 2:
+      return "SHA256"
+    elif cs_hash_type == 3:
+      return "SHA256_TRUNCATED"
+    elif cs_hash_type == 4:
+      return "SHA384"
+    else:
+      return ""
+
+  def _ParseCodeSignature(self, code_signature, result):
+    """Parses Mach-O code signature.
+       Details about the code signature structure on GitHub
+       https://github.com/apple-oss-distributions/xnu/blob/94d3b452840153a99b38a3a9659680b2a006908e/osfmk/kern/cs_blobs.h#L267
+    Args:
+      code_signature (lief.MachO.CodeSignature): code signature to be parsed.
+      result (TurbiniaTaskResult): The object to place task results into.
+    Returns: Signature or None if not found
+    """
+    signature = None
+    signature_bytes = code_signature.content.tobytes()
+    super_blob_magic = signature_bytes[0:4]  # uint32_t magic
+    if super_blob_magic == self._CSMAGIC_EMBEDDED_SIGNATURE:
+      # SuperBlob called EmbeddedSignatureBlob found which contains the code signature data
+      # struct EmbeddedSignatureBlob {
+      #   uint32_t magic = 0xfade0cc0;
+      #   uint32_t length;
+      #   uint32_t count; // Count of contained blob entries
+      #   IndexEntry entries[]; // Has `count` entries
+      #   Blob blobs[]; // Has `count` blobs
+      # }
+      signature = Signature(signer_infos=[])
+      result.log(f'_CSSLOT_SIGNATURESLOT: {self._CSSLOT_SIGNATURESLOT}')
+      super_blob_length = int.from_bytes(
+          signature_bytes[4:8], "big")  # uint32_t length
+      generic_blob_count = int.from_bytes(
+          signature_bytes[8:12],
+          "big")  # uint32_t count: Count of contained blob entries
+      result.log(f'super_blob_length: ' + str(super_blob_length))
+      result.log(f'generic_blob_count: ' + str(generic_blob_count))
+      for i in range(generic_blob_count):
+        # lets walk through the CS_BlobIndex index[] entries
+        # https://github.com/apple-oss-distributions/xnu/blob/94d3b452840153a99b38a3a9659680b2a006908e/osfmk/kern/cs_blobs.h#L280C15-L280C22
+        start_index_entry_type = 12 + i * 8  # uint32_t type
+        start_index_entry_offset = start_index_entry_type + 4  # uint32_t offset
+        result.log(f' start_type:  {start_index_entry_type}')
+        result.log(f' start_index_entry_offset: {start_index_entry_offset}')
```

Comment:
Minor nit:  I understand that these log lines are indented because it's in a loop and there will be multiple of them, but there are a few other log lines below in the same section that have varying levels if indentation.  Is there a reason they're not at a single indent similar to the above line?

Reply:
This output was for development only. Will remove as it is no longer required.